management-node/server.go

```diff
@@ -0,0 +1,530 @@
+	log.Info("WorkerConnect is called")
+
+	// TBD NewWorkerNode func style
+	// creating worker node that has connected
+	wRPCClient := wrpc.NewWorkerNodeRPCClient("", stream, &pb.WorkerNode{
+		Id:      "",
+		Address: "TBD",
+		State:   common.WorkerNodeStateConnected,
+	}, Config.SilenceTimeout)
+
+	// by architecture decision worker connects and send first
+	// message to via stream
+	// message have to contain its id in reply field
+	msg, err := wRPCClient.RxWithTimeout()
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	rsp := msg.(*pb.WorkerRsp)
+	wRPCClient.SetId(rsp.Reply)
+
+	err = mn.AddWorkerNode(wRPCClient)
+	if err != nil {
+		log.Error(err.Error())
+		return nil
+	}
+
+	// to make async communication between Worker and Manager
+	// because sending via stream is possible only synchronously
+	wRPCClient.StartSender()
+
+	// launch pinger to check connection state
+	wRPCClient.StartPinger(time.Duration(Config.PingTimer))
+
+	// launch main eventloop
+	if err := wRPCClient.InitLoop(); err != nil {
+		log.Error(err.Error())
+	}
+
+	wRPCClient.StopPinger()
+	wRPCClient.StopSender()
+
+	rmErr := mn.RmWorkerNode(wRPCClient)
+	if rmErr != nil {
+		//concat errors in case there are problems with db
+		err = fmt.Errorf("%s %s", err.Error(), rmErr.Error())
+		log.Error(err.Error())
+	}
+
+	return err
+}
+
+// GetWorkerList gRPC that is called by schedctl with parameters
+// of DummyReq (for now). In future can be used for proper querying
+// and returns list of workers that reside in db
+// TBD: can be done with pagination and limiting on rsp
+//		or even changed to stream
+func (md *ManagementNode) GetWorkerList(ctx context.Context, req *pb.DummyReq) (*pb.WorkerNodeList, error) {
+
+	log.Info("GetWorkerList")
+	// for now it gets all recorded workers
+	opts := []clientv3.OpOption{
+		clientv3.WithPrefix(),
+	}
+
+	eCtx, cancel := context.WithTimeout(context.Background(),
+		time.Duration(Config.DialTimeout)*time.Second)
+	gr, err := md.etcd.Get(eCtx, EtcdWorkerPrefix, opts...)
+	cancel()
+	if err != nil {
+		log.Error(err.Error())
+		return nil, err
+	}
+
+	wnl := pb.WorkerNodeList{}
+
+	for _, item := range gr.Kvs {
+		var wn wrpc.WorkerRPCClient
+		err := json.Unmarshal([]byte(item.Value), &wn)
+		if err != nil {
+			log.Error(err.Error())
+			return nil, err
+		}
+		wnl.Nodes = append(wnl.Nodes, wn.WN)
+	}
+
+	return &wnl, nil
+}
+
+// GetTaskList is gRPC that is called by schedctl with parameters
+// of DummyReq (for now). In future can be used for proper querying
+// and returns list of tasks that reside in db
+// TBD: can be done with pagination and limiting on rsp
+//		or even changed to stream
+func (md *ManagementNode) GetTaskList(ctx context.Context, req *pb.DummyReq) (*pb.TaskList, error) {
+
+	log.Info("GetTaskList")
+	opts := []clientv3.OpOption{
+		clientv3.WithPrefix(),
+	}
+
+	eCtx, cancel := context.WithTimeout(context.Background(),
```

Comment:
Use context passed to this methods as parent?

Reply:
done